packages/corporate-integration/src/HypernetCorporateIntegration.ts

```diff
@@ -0,0 +1,144 @@
+import {
+	UUID,
+	AjaxError,
+	JsonWebToken,
+	EmailAddressString,
+	EthereumAccountAddress,
+} from "@hypernetlabs/objects";
+import { IAjaxUtils, AxiosAjaxUtils } from "@hypernetlabs/utils";
+import { apiBaseUrl } from "@corporate-integration/configuration";
+import {
+	IHypernetCorporateIntegration,
+	NFTMetadata,
+} from "@corporate-integration/IHypernetCorporateIntegration";
+
+import { injectable } from "inversify";
+import { okAsync, ResultAsync } from "neverthrow";
+import { Readable } from "stream";
+import jwt_decode, { JwtPayload } from "jwt-decode";
+
+@injectable()
+export class HypernetCorporateIntegration
+	implements IHypernetCorporateIntegration
+{
+	protected ajaxUtils: IAjaxUtils;
+	protected token: string | null = null;
+	protected tokenExpiration: number | undefined;
+
+	constructor(
+		protected corporateId: UUID,
+		protected corporateSecret: string,
+	) {
+		this.ajaxUtils = new AxiosAjaxUtils();
+	}
+
+	public uploadFileToCollection(
+		collectionId: UUID,
+		fileContent: Buffer,
+		fileName: string,
+	): ResultAsync<void, AjaxError> {
+		return this.setCorporateAuthenticationToken().andThen(() => {
+			return this.getUploadLink(collectionId, fileName)
+				.andThen((url) => {
+					return this.uploadFileToBucket(url, fileContent);
+				})
+				.map(() => {});
+		});
+	}
+
+	public createNFTInCollection(
+		collectionId: UUID,
+		metadata: NFTMetadata,
+		identityId: UUID | null,
+		emailAddress: EmailAddressString | null,
+		accountAddress: EthereumAccountAddress | null,
+		imageFileame: string | null,
+		animationFilename: string | null,
+	): ResultAsync<UUID, AjaxError> {
+		return this.setCorporateAuthenticationToken().andThen(() => {
+			const requestUrl = new URL(
+				`${apiBaseUrl}/collections/${collectionId}/nfts`,
+			);
+			return this.ajaxUtils
+				.post<{ nftId: UUID }>(requestUrl, {
+					metadata,
+					identityId,
+					emailAddress,
+					accountAddress,
+					imageFileame,
+					animationFilename,
+				})
+				.map((result) => {
+					return result.nftId;
+				});
+		});
+	}
+
+	protected setCorporateAuthenticationToken(): ResultAsync<void, AjaxError> {
+		if (
+			this.token == null ||
+			(this.token != null &&
+				(this.tokenExpiration == null ||
+					this.tokenExpiration < Date.now()))
```

Comment:
Make sure to do something like tokenExpiration - 10 minutes < Date.now() so that we prematurely refresh the token.